### PR TITLE
chore: migrate to module

### PR DESCRIPTION
**Pull request by jobo322**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { resolve } from 'path';
+
+import { fileListFromPath } from 'filelist-utils';
+
+const path = resolve('../data/');
+const fileList = fileListFromPath(path);
+
+export const getList = () => fileList.map((d) => d.name);
```
Comment (lpatiny) [inline src/index.js:8]: Why not writing 
```
export function getList() {

}
```
This is our prefer way seems to me.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const result = fileList.filter((d) => d.name === name);
+
+  if (result.length < 1) {
+    throw new Error(`There is not a file with name: ${name}`);
+  }
+
+  return result[0];
+};
+
+export const getData = (name) => {
```
Comment (lpatiny) [inline src/index.js:20]: same here

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { resolve } from 'path';
+
+import { fileListFromPath } from 'filelist-utils';
+
+const path = resolve('../data/');
+const fileList = fileListFromPath(path);
+
+export const getList = () => fileList.map((d) => d.name);
+
+export const getFileList = (name) => {
```
Comment (lpatiny) [inline src/index.js:10]: same here

File: package.json [hunk 4]
```diff
@@ -2,27 +2,51 @@
...
+  "homepage": "https://github.com/cheminfo/jeol-data-test#readme",
+  "prettier": {
+    "arrowParens": "always",
+    "semi": true,
+    "singleQuote": true,
+    "tabWidth": 2,
+    "trailingComma": "all"
+  },
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 0"
```
Comment (lpatiny) [inline package.json:21]: Could you add a simple test so that we can use continuous integration.

**Comment by lpatiny** [comment]
package-lock should not exists. .npmrc with `package-lock=false` should be added

**Review by lpatiny — COMMENTED** [review]
Please could you also check that lib and lib-esm are in .gitignore

**Comment by targos** [comment]
Some comments/explanations about https://github.com/cheminfo/jeol-data-test/pull/2/commits/8960de7fe9d1f11c9e8b2613e58ca984f68c33c5:

## ESLint error

image

This error is fixed by:

image

The reason for that error is not related to ESM, as you might think. It's because recent versions of our ESLint config consider by default that we are in an environment compatible with web browsers, and web browsers do not have the `__dirname` variable. For projects like this one, which are meant to be used in Node.js only, you can add the "env: node: true" config in ESLint and then it knows that `__dirname` and other things like `require` or `process` are available.

## `url` module

image

The `url` module is a Node.js builtin, so there is no need to install it. In fact, the version installed by npm will never be loaded by Node.

## ESM usage

image

For the past years, and still today in most projects, we still don't really use the standard ESM syntax. We write modules with import/export but they are always transpiled before publication (with Rollup) and when running tests (with Jest/babel).
The rules in projects which have a babel config with `@babel/plugin-transform-modules-commonjs` are:
- Avoid `require`. Use import/export in source code and in tests
- Do not use `import.meta.url` (it only works in native ESM)
- You can always use `__dirname` in tests, and optionally in source code if you set the ESLint env like explained above.

**Pull request merged**